src/main/java/org/openstreetmap/atlas/checks/validation/areas/PedestrianAreaOverlappingEdgeCheck.java

```diff
@@ -0,0 +1,245 @@
+                final boolean intersectsAtEnd = locationsInSegments
+                        .contains(edgeSegmentEndLocation);
+                // If both ends of the edge intersects the area, check if it is properly snapped by
+                // checking if both the intersecting points are on the same segment.
+                // If none of the filtered segments have both the start and end nodes, then add the
+                // edge and all its connected edges that are within the area to flag them.
+                if ((intersectsAtStart && intersectsAtEnd
+                        && this.isNotSnappedToSegments(segments, segment))
+                        // If any one of the end of the connected edge is fully enclosed within the
+                        // area, flag the edge and all its connected edges that are within the area.
+                        || (intersectsAtStart && !intersectsAtEnd
+                                && DISTANCE_GREATER_THAN_ONE_METER.test(endLocDistance)
+                                && areaPolygon.fullyGeometricallyEncloses(edgeSegmentEndLocation))
+                        || (intersectsAtEnd && !intersectsAtStart
+                                && DISTANCE_GREATER_THAN_ONE_METER.test(startLocDistance)
+                                && areaPolygon
+                                        .fullyGeometricallyEncloses(edgeSegmentStartLocation)))
+                {
+                    // Collect all connected edges that are within the pedestrian area
+                    edge.connectedEdges().stream().filter(connectedEdge -> this
+                            .isValidIntersectingEdge(connectedEdge, area)
+                            && areaPolygon.fullyGeometricallyEncloses(connectedEdge.asPolyLine()))
+                            .forEach(overlappingEdges::add);
+                    // Add the intersecting edge as well to the set
+                    overlappingEdges.add(edge);
+                }
+            });
+        }
+        if (!overlappingEdges.isEmpty())
+        {
+            this.markAsFlagged(object.getOsmIdentifier());
+            final CheckFlag flag = this.createFlag(overlappingEdges,
+                    this.getLocalizedInstruction(0, object.getOsmIdentifier(),
+                            new StringList(getIdentifiers(overlappingEdges)).join(", ")));
+            flag.addObject(object);
+            return Optional.of(flag);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * Collects all atlas identifiers of given set of {@link AtlasObject}s
+     *
+     * @param objects
+     *            set of {@link AtlasObject}s
+     * @return {@link Iterable<String>} containing the atlas identifiers of input objects
+     */
+    private Iterable<String> getIdentifiers(final Set<AtlasObject> objects)
```

Comment:
This seems like it could be useful in a super class of PAOEC (in the future)

Reply:
Agree that it could be useful to have this method publicly accessible. I will take a look at which Atlas class this can go in.